ogr/ogrsf_frmts/hana/ogrhanafeaturereader.cpp

```diff
@@ -63,12 +63,18 @@ odbc::String CreateStringFromValues(
 template<typename T>
 T castInt(int value)
 {
-    if (!(std::numeric_limits<T>::min() >= value
-          || value <= std::numeric_limits<T>::max()))
+    if (!(value >= std::numeric_limits<T>::min() &&
+          value <= std::numeric_limits<T>::max()))
```

Comment:
Wouldn't this be a bit clearer? (Or is the ! condition used to handle some corner case with nan values or suchlike?)

```suggestion
    if ( value < std::numeric_limits<T>::min() ||
          value > std::numeric_limits<T>::max() )
```

Reply:
changed to your suggestion. It is only used with T=int8_t and int16_t